deep_quoridor/src/quoridor.py

```diff
@@ -0,0 +1,293 @@
+
+            # Destination cell must be free
+            is_valid = True
+            if self.board.get_cell(destination_position) != Board.FREE:
+                is_valid = False
+
+            elif np.sum(np.abs(position_delta)) == 1:
+                # Moving to an adjacent cell, just make sure no wall is in the way.
+                if self.board.is_wall_between(current_position, destination_position):
+                    is_valid = False
+
+            elif (tuple(opponent_offset), tuple(position_delta)) in self._jump_checks:
+                jump_checks = self._jump_checks[(tuple(opponent_offset), tuple(position_delta))]
+
+                for check_delta_1, check_delta_2 in jump_checks["wall"]:
+                    if not self.board.is_wall_between(
+                        current_position + check_delta_1, current_position + check_delta_2
+                    ):
+                        is_valid = False
+                for check_delta_1, check_delta_2 in jump_checks["nowall"]:
+                    if self.board.is_wall_between(current_position + check_delta_1, current_position + check_delta_2):
+                        is_valid = False
+            else:
+                # This isn't a move by 1, and it isn't a jump, so it's invalid.
+                is_valid = False
+
+        elif isinstance(action, WallAction):
+            is_valid = self.board.can_place_wall(action.position, action.orientation)
+
+        else:
+            raise ValueError("Invalid action type")
+
+        return is_valid
+
+    def set_current_player(self, player: int):
+        if player not in [0, 1]:
+            raise ValueError(f"Invalid player {player}")
+        self.current_player = player
+
+    def get_current_player(self) -> int:
+        return self.current_player
+
+    def __str__(self):
+        return str(self.board)
+
+
+def create_jump_checks():
+    """
+    Pre-generate a sequence of checks for possible jumps.
+
+    Pre-generating the checks saves some computation time checking validity of move actions later.
+
+    There are 2 types of checks:
+    - nowall: There must not be a wall between each of these pairs of positions.
+    - wall: There must be a wall between each of these pairs of positions.
+
+    Positions are relative to the player's current position.
+    """
+    # Checks for the case where we are moving to (row + 1, col) or jumping over
+    # an opponent at (row + 1, col)
+    checks_for_one_direction = {
+        (2, 0): {
+            "wall": [],
+            "nowall": [((0, 0), (1, 0)), ((1, 0), (2, 0))],
+        },
+        (1, 1): {
+            "wall": [((1, 0), (2, 0))],
+            "nowall": [((0, 0), (1, 0)), ((1, 0), (1, 1))],
+        },
+        (1, -1): {
+            "wall": [((1, 0), (2, 0))],
+            "nowall": [((0, 0), (1, 0)), ((1, 0), (1, -1))],
+        },
+    }
+    # Rotate the checks for all 4 possible directions and convert to numpy arrays.
+    checks = {}
```

Comment:
This is slick but also a bit hard to read. Provided it is well tested on this function we are. 👍

Reply:
Yeaaaaah. I tried a bunch of different ways to check jumps, and all of them were hard to read. This is a big candidate for refactor at a later date if someone gets a good idea :-)